From 30dcb9e2d1af10290d0764860b47b6f4848a1b93 Mon Sep 17 00:00:00 2001
From: SolidHal <hal@halemmerich.com>
Date: Wed, 23 Dec 2020 10:03:26 -0800
Subject: [PATCH 1/1] bridge/analogix: Don't wait for panel ACK on PSR exit

We currently wait for the panel to mirror our intended PSR state
before continuing on both PSR enter and PSR exit. This is really
only important to do when we're entering PSR, since we want to
be sure the last frame we pushed is being served from the panel's
internal fb before shutting down the soc blocks (vop/analogix).

This patch changes the behavior such that we only wait for the
panel to complete the PSR transition when we're entering PSR, and
to skip verification when we're exiting.

Without this, the system essentially freezes for ~100ms while it
waits for confirmation that PSR is disabled. The most noticible
behavior is the cursor jumping on quick inputs. 

from: https://chromium.googlesource.com/chromiumos/third_party/kernel/+/e47a7da072d1a2ca8fdc62f3e32291c0d1a41145%5E%21/#F0
Signed-off-by: Hal Emmerich <hal@halemmerich.com>
---
 drivers/gpu/drm/bridge/analogix/analogix_dp_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c b/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c
index aa1bb86293fd..163bc069f77a 100644
--- a/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c
+++ b/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c
@@ -1055,7 +1055,7 @@ static int analogix_dp_disable_psr(struct analogix_dp_device *dp)
 	psr_vsc.db[0] = 0;
 	psr_vsc.db[1] = 0;
 
-	return analogix_dp_send_psr_spd(dp, &psr_vsc, true);
+	return analogix_dp_send_psr_spd(dp, &psr_vsc, false);
 }
 
 /*
-- 
2.20.1

